File: app/src/main/java/seigneur/gauvain/mycourt/ui/pin/PinActivityModule.java
package seigneur.gauvain.mycourt.ui.pin;

import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import seigneur.gauvain.mycourt.di.scope.PerActivity;

/**
 * Provides activity dependencies.
 */
@Module
public abstract class PinActivityModule {

    @Binds
    @PerActivity
    abstract Activity activity(PinActivity pinActivity);


}

File: app/src/main/java/seigneur/gauvain/mycourt/ui/pin/PinActivityModule.kt
package seigneur.gauvain.mycourt.ui.pin

import android.app.Activity

import dagger.Binds
import dagger.Module
import seigneur.gauvain.mycourt.di.scope.PerActivity

/**
 * Provides activity dependencies.
 */
@Module
abstract class PinActivityModule {

    @Binds
    @PerActivity
    internal abstract fun activity(pinActivity: PinActivity): Activity


}